### PR TITLE
[Playwright] General improvement

### PR DESCRIPTION
**Pull request by aweng98**

### CODE REVIEW

File: playwright-e2e/tests/dsm/kitUploadFlow/blood-kit-prefix-check.spec.ts [hunk 1]
```diff
@@ -141,14 +140,7 @@ test.describe.serial('Blood Kit Upload', () => {
...
 
         await kitsTable.selectSingleRowByIndex();
-        await Promise.all([
-          waitForResponse(page, { uri: '/kitLabel' }),
-          kitsWithoutLabelPage.createLabelsButton.click()
-        ]);
-        await Promise.all([
-          expect(page.locator('[data-icon="cog"]')).toBeVisible({timeout: 60000}),
-          expect(page.locator('h3')).toHaveText(/Triggered label creation/i, {timeout: 60000})
-        ]);
```
Comment (aweng98) [inline playwright-e2e/tests/dsm/kitUploadFlow/blood-kit-prefix-check.spec.ts:151]: Duplicated code, already in `clickCreateLabels()`.

File: playwright-e2e/tests/dsm/kitUploadFlow/blood-and-rna-kit-upload-flow.spec.ts [hunk 2]
```diff
@@ -122,15 +121,7 @@ test.describe('Blood & RNA Kit Upload', () => {
...
     await kitsTable.selectSingleRowByIndex();
-    await Promise.all([
-      waitForResponse(page, { uri: '/kitLabel' }),
-      kitsWithoutLabelPage.createLabelsButton.click()
-    ]);
-    await Promise.all([
-      expect(page.locator('[data-icon="cog"]')).toBeVisible(),
-      expect(page.locator('h3')).toHaveText(/Triggered label creation/i)
-    ]);
-    await kitsWithoutLabelPage.waitUntilAllKitLabelCreationRequestsAreProcessed();
```
Comment (aweng98) [inline playwright-e2e/tests/dsm/kitUploadFlow/blood-and-rna-kit-upload-flow.spec.ts:133]: Duplicated code, already in `clickCreateLabels()`.

File: playwright-e2e/dsm/pages/dsm-page-base.ts [hunk 3]
```diff
@@ -11,7 +11,10 @@ export default abstract class DsmPageBase {
   }
 
   async reload(): Promise<void> {
+    const networkIdlePromise = this.page.waitForLoadState('networkidle');
```
Comment (aweng98) [inline playwright-e2e/dsm/pages/dsm-page-base.ts:14]: Use this instead `page.reload()` for handling UI synchronization.

**Review by Rinchan20 — APPROVED** [review]

File: playwright-e2e/dsm/pages/participant-list-page.ts [hunk 4]
```diff
@@ -247,20 +248,19 @@ export default class ParticipantListPage {
...
       }
-      if (!allowNewYorkerOrCanadian &&
-        ((country === 'CA') ||
-        ((country === 'US') && (state === 'NY')))) {
+      if (!allowNewYorkerOrCanadian && (country === 'CA' || (country === 'US' && state === 'NY'))) {
         return -1;
       }
+      if (matchFirstName && normalCollaboratorSampleID.length <= 5 && isAddressValid.toLowerCase().trim() === 'true') {
+        return index;
+      }
```
Comment (aweng98) [inline playwright-e2e/dsm/pages/participant-list-page.ts:263]: @Rinchan20  Two if-condition checks were in the wrong order. It led to "bad" (unsuitable) participants.

```
if (!allowNewYorkerOrCanadian && (country === 'CA' || (country === 'US' && state === 'NY')))
```
and 
```
if (matchFirstName && normalCollaboratorSampleID.length <= 5 && isAddressValid.toLowerCase().trim() === 'true')
```

**Pull request merged**